File: Remove Element/Remove Element.java
public class Solution {
    public int removeElement(int[] A, int elem) {
        int swapPtr = A.length - 1;
        if(swapPtr < 0){
            return 0;
        }
        while(A[swapPtr] == elem){
            --swapPtr;
            if(swapPtr < 0){
                break;
            }
        }
        for(int i = 0; i <= swapPtr;){
            if(A[i] == elem){
                int temp = A[i];
                A[i] = A[swapPtr];
                A[swapPtr] = temp;
                --swapPtr;
            }
            else{
                ++i;
            }
        }
        return swapPtr + 1;
    }
}

File: Remove Element/Remove Element.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {

        for(unsigned int i = 0; i < n;){
            // Find element to be removed
            if(A[i] == elem){
                // Remove element at index i
                // Overwrite A[i] with A[i + 1] and so on
                for(unsigned int j = i + 1; j < n; ++j){
                    A[j - 1] = A[j];
                }
                // Length is shortened
                n--;
            }
            // Move index
            else{
                ++i;
            }
        }
        return n;
    }
};
